es.go

```diff
@@ -767,9 +767,28 @@ func (c *Client) ModifyAliases(actions []AliasAction) error {
 //
 // Use case: You want to remove an index and all of its data.
 func (c *Client) DeleteIndex(indexName string) error {
+	return c.DeleteIndexWithQueryParameters(indexName, nil)
+}
+
+// Delete an index in the cluster with query parameters.
+//
+// Use case: You want to remove an index and all of its data. You also want to
+// specify query parameters such as timeout.
+func (c *Client) DeleteIndexWithQueryParameters(indexName string, queryParamMap map[string][]string) error {
+	queryParams := make([]string, len(queryParamMap))
```

Comment:
here we are initializing the slice with `len(queryParamMap)` strings already inside. What we should do is set 0 as length and l`en(queryParamMap)` as the capacity `queryParams := make([]string, 0, len(queryParamMap))`

Here a little example of what I'm talking about https://play.golang.com/p/QZOciUHsI1n

Reply:
Ahh, yes. That's something I missed. Thanks a lot!